python/ray/llm/_internal/batch/stages/sglang_engine_stage.py

```diff
@@ -0,0 +1,367 @@
+"""The stage that runs SGLang engine."""
+
+import asyncio
+import logging
+import time
+import uuid
+from enum import Enum
+from pydantic import BaseModel, root_validator
+from typing import Any, Dict, AsyncIterator, Optional, List, Tuple, Type
+
+from ray.llm._internal.utils import try_import
+from ray.llm._internal.batch.stages.base import (
+    StatefulStage,
+    StatefulStageUDF,
+)
+
+sgl = try_import("sglang")
+
+logger = logging.getLogger(__name__)
+
+
+class SGLangTaskType(str, Enum):
+    """The type of task to run on the SGLang engine."""
+
+    """Generate text."""
+    GENERATE = "generate"
+
+    # TODO: Add support for the following tasks.
```

Comment:
We can only support Generate for now and add these when they are actually supported.

Reply:
Thanks for pointing this out! Initially, I thought including these components might make this PR too large and hard to review, so I left them out intentionally. But if you prefer that everything is included for a complete end-to-end flow, I'd be more than happy to add the missing parts you mentioned.